src/charm.py

```diff
@@ -930,6 +974,20 @@ def _generate_middleware_config(
           "cannot create middleware: multi-types middleware not supported, consider declaring two
           different pieces of middleware instead"
         """
+        forwardauth_middleware = {}
+        if self.forward_auth.is_ready():
+            forward_auth_config = self.forward_auth.get_provider_info()
+            if data.get("name") in forward_auth_config.app_names:  # type: ignore
```

Comment:
Don't quite follow what this condition does. Could you please explain?

Reply:
Sure. The `_generate_middleware_config` method generates middlewares for each charm related with one of the ingress interfaces.
In this block I want to check if the app name, for which the middlewares are currently generated (`data.get("name")`), requested IAP protection (`forward_auth_config.app_names`). If so, the `forwardAuth` definition should be added.
Let me know if it's clearer now.

Reply:
Thanks!
Should this logic (including `is_ready`?) be owned by forward_auth lib? E.g.
```python
if self.forward_auth.is_protected_app(data.get("name")):
```

Reply:
I'm not sure we should move all the logic to the lib, especially given that the `forward_auth.is_ready():` check is required in a few blocks anyways

Reply:
It's fine to keep `is_ready` public.
It seems that the flow `is_ready -> get_provider_info -> app_name in config.app_names` requires some know-how about how the lib is built, and could potentially be accomplished by a lib method?